### ISSUE TITLE
Accuracy testing for Inductor

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#826]
A lot of models are failing accuracy checks for float16 and AMP. It could be just minor deviations where we could relax accuracy tests or it could be hiding real bugs. We should do a cleanup.

**Comment by jansel** [comment]
Are we comparing to float64 ground truth?  Or eager float16?

**Comment by anijain2305** [comment]
Eager float16. Happy to discuss ways to make this simpler.

**Comment by jansel** [comment]
Some differences might be expected here.  The accuracy bar should be: "not significantly further away from the answer float64 precision would give when compared to eager"

**Comment by anijain2305** [comment]
Adding more details to this task

* Use deviations from fp64 to measure the sameness of two tensors.
* Revisit if we can get rid of no-decomp version, for TIMM models.
* Inductor skips accuracy checks for now. This is bad. We want to use model.train(). Try, (1) Restoring rng state, or (2) @Chillee suggested to use TorchDispatcher to replace randn with a factory function.

**Comment by anijain2305** [comment]
Update - First 2 tasks are done.

Remaining task is to do better accuracy testing for Inductor.

**Comment by voznesenskym** [comment]
Do we compare that NE stays the same for training across models with different degrees of quantization?

**Comment by anijain2305** [comment]
What is NE? We do not test quantization at all.

**Comment by ezyang** [comment]
normalized entropy; it's how most of the prod rec models figure out how good the model is doing.

**Comment by voznesenskym** [comment]
> What is NE? We do not test quantization at all.

isn't our fp16/fp32/fp64 flags quantization? Maybe I used the wrong word... what do we call those modes?

**Comment by voznesenskym** [comment]
> normalized entropy; it's how most of the prod rec models figure out how good the model is doing.

Yea. It doesn't have to be NE - any one number representation suffices, NE is just a good one.

**Comment by anijain2305** [comment]
Done
